### PR TITLE
RFC-13: Add Gateway Pair

### PR DESCRIPTION
**Pull request by zu1k**

### CODE REVIEW

File: docs/rfcs/0-example.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+- Author: (fill me in with `name <mail>`, e.g., Xuanwo <github@xuanwo.io>)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [beyondstorage/beyond-fs#0](https://github.com/beyondstorage/beyond-fs/issues/0)
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]: Please update the template.

File: docs/rfcs/0-example.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- Tracking Issue: [beyondstorage/beyond-fs#0](https://github.com/beyondstorage/beyond-fs/issues/0)
```
Comment (Xuanwo) [inline docs/rfcs/0-example.md]: ditto

File: docs/rfcs/1-add-pair-for-reacher.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+- Author: zu1k <i@lgf.im>
+- Start Date: 2021-07-15
+- RFC PR: [beyondstorage/go-service-ipfs#13](https://github.com/beyondstorage/go-service-ipfs/pull/13)
+- Tracking Issue: [beyondstorage/go-service-ipfs#14](https://github.com/beyondstorage/go-service-ipfs/issues/14)
+
+# RFC-1: Add a pair for `Reacher`
```
Comment (Xuanwo) [inline docs/rfcs/1-add-pair-for-reacher.md]: How about using PR numbers as the RFC ID?

- Following other beyondstorage's project style
- Easy to identify
- Avoid conflict

Comment (Xuanwo) [reply docs/rfcs/1-add-pair-for-reacher.md]: Maybe we can make the RFC more readable by changing the name into `Add Gateway Pair`?

File: docs/rfcs/1-add-pair-for-reacher.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Background
+
+With IPFS, we can get the `CID` of an object, then we can use it with a public or internal IPFS Gateway to splice out the access link.
+
+When using internal gateway like `127.0.0.1:8080`, the access link are inaccessible from external network.
+
+When using a public gateway, e.g. `ipfs.io`, `cf-ipfs.com`, if the service is an intranet IPFS cluster, it is possible that the files are not accessible from these public gateway as well.
+
+Releated issue: [beyondstorage/go-service-ipfs#5](https://github.com/beyondstorage/go-service-ipfs/issues/5)
```
Comment (Xuanwo) [inline docs/rfcs/1-add-pair-for-reacher.md]: Maybe we can move this section just before `## Background`?

File: docs/rfcs/1-add-pair-for-reacher.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+### How to splice out the access link?
+
+- https://{gateway URL}/ipfs/{CID}/{optional path to resource}
+- https://{CID}.ipfs.{gatewayURL}/{optional path to resource}
+
+## Proposal
+
+I propose to add a pair to let the user specify the `gateway`.
```
Comment (Xuanwo) [inline docs/rfcs/1-add-pair-for-reacher.md]: Can you provide more details?

- What's the gateway pair's type?
- What's the gateway pair's value? Will we use the `go-endpoint` or just plain url?
- Where can we use the pair? In `Reach` operation only?

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
ping @beyondstorage/go-storage-maintainer @beyondstorage/go-storage-committer for another LGTM

**Review by xxchan — APPROVED** [review]

**Review by xxchan — APPROVED** [review]

**Pull request merged**